### ISSUE TITLE
Limit pipeline nesting to one level

### ISSUE THREAD

**Issue by binarylogic** [vectordotdev/vector#9991]
The `pipeline` transform allows for nested `pipelines` transforms. This is necessary to support Datadog's nested Pipelines feature, but we only want to allow one level of nested pipelines. This is because:

1. The UI will only support a single level.
2. We want to limit complexity. The purpose of this transform is to facilitate collaboration by limiting complexity.

Instead, pipelines should support nested pipelines, but one level. Because we are already in the context of a specific pipeline, the type of the nested pipeline should be elided.

Example configuration:

```toml
[transforms.processing.logs.pipelines.services]
name = "Services"
filter.type = "datadog_search"
filter.condition = "service:*"

order = [
    "auth",
    "api"
]

[transforms.processing.logs.pipelines.services.pipelines.auth]
name = "Authentication Service"
filter.type = "datadog_search"
filter.condition = "service:auth"

[[transforms.processing.logs.pipelines.services.pipelines.auth.transforms]]
type = "remap"
source = '' #...

[transforms.processing.logs.pipelines.services.pipelines.api]
name = "API Service"
filter.type = "datadog_search"
filter.condition = "service:api"

[[transforms.processing.logs.pipelines.services.pipelines.api.transforms]]
type = "remap"
source = '' #...
```

**Comment by leebenson** [comment]
To clarify the expected UX: Should additional levels cause a runtime error (hard exit) of Vector?

Or would we simply ignore additional levels and issue a warning?

**Comment by binarylogic** [comment]
Hard exit. I'd like to treat it like any other configuration error.

**Comment by vladimir-dd** [comment]
As we agreed offline, we should forbid nested pipelines completely. We can "flatten" nested DD pipelines. Closing this issue.

**Comment by jszwedko** [comment]
@vladimir-dd I actually updated the description here to reflect the new directly embedded pipelines (i.e. directly configured rather than a nested `pipelines` transform). Do we not need these either? Can you show an example of how they would be flattened?

**Comment by binarylogic** [comment]
@jszwedko I spoke to @vladimir-dd about this. In general, I dislike the nesting of pipelines as a feature since we allow multiple `pipelines` transforms for better organization of pipelines. I'd prefer to learn from users if they want this given that existing flexibility.

**Comment by vladimir-dd** [comment]
Here is an [example](https://github.com/DataDog/vector-enterprise/blob/vladimir-dd/outreach-config/vector.toml#L308) of flattened pipelienes(including 2 levels deep)

**Comment by jszwedko** [comment]
Thanks @vladimir-dd . I touched base with @binarylogic about this who filled me in more. The above comment he left summarizes it.

Just for the future, when closing issues because of outside discussion, it is useful to leave more context on the issue itself so that other readers, not part of the discussion, can follow along. Also any of us 12-24 months down the line when we've forgotten why we made the decision 😄
